### PR TITLE
Simplify `extr_neighbor`

### PR DESCRIPTION
**Pull request by AnthonyMikh**
This PR also changes the interface of function, but it should be fine since it is not public and it is used only trough `low_neighbor` and `high_neighbor`. New implementation passes all the tests in this module.

### CODE REVIEW

File: src/audio.rs [hunk 1]
```diff
@@ -255,41 +255,45 @@ fn floor_one_decode(rdr :&mut BitpackCursor, codebooks :&[Codebook],
...
-	return (extr_idx, max_val);
+	// the criterion of being "smaller" than bound
+	let min_idx = prefix.iter()
+		.position(|&val| smaller(val, bound))
+		.unwrap_or_else(|| 
+			panic!("No index y < {0} found where v[y] is {1} than v[{0}] = 0x{2:08x}!",
+				max_idx, relation, bound));
+
+	// Now search for "bigger" entries
+	let (offset, max_neighbor) = prefix[min_idx..].iter().cloned()
```
Comment (est31) [inline src/audio.rs]: Why do you do cloned here?

Comment (AnthonyMikh) [reply src/audio.rs]: In order to iterate over `u32`s rather than `&u32`s. For such a small type there shouldn't be any performance penalty.

Comment (est31) [reply src/audio.rs]: Thanks! That makes sense.

Comment (Shnatsel) [reply src/audio.rs]: You can actually use `.copied()` to make this more explicit

Comment (est31) [reply src/audio.rs]: Good point. copied is fairly recent addition tho, don't want to require such a new version of the language.

**Pull request merged**